File: Problems/0977_Squares_of_a_Sorted_Array/Project_Python3/Squares_of_a_Sorted_Array.py
# coding: utf-8

import os
import sys
import time

class Solution:
    def sortedSquares(self, A):
        """
        :type A: List[int]
        :rtype: List[int]
        """
        squareA = [0]*len(A)
        for i in range(len(A)):
            squareA[i] = A[i]**2
        squareA.sort()
        return squareA

def str_to_int_array(flds):
    if len(flds) <= 0:
        return None
    temp = flds.split(",")
    nums = [0]*len(temp)
    for i in range(len(temp)):
        nums[i] = int(temp[i])
    return nums

def main():
    argv = sys.argv
    argc = len(argv)

    if (argc < 2):
        print("Usage: python %s <testdata.txt>" %(argv[0]))
        exit(0)

    if not os.path.exists(argv[1]):
        print("%s not found..." %argv[1])
        exit(0)

    testDataFile = open(argv[1], "r")
    lines = testDataFile.readlines()

    for temp in lines:
        print("argv[1] = %s" %temp)
        loop_main(temp)
    #    print("Hit Return to continue...")
    #    input()

def loop_main(temp):
    flds = temp.replace("[","").replace("]","").rstrip()
    A = str_to_int_array(flds)

    time0 = time.time()

    sl = Solution()
    result = sl.sortedSquares(A)
    print("result = %s" %result)

    time1 = time.time()

    print("Execute time ... : %f[s]" %(time1 - time0))
    print()

if __name__ == "__main__":
    main()

File: Problems/0900_0999/0977_Squares_of_a_Sorted_Array/Project_CS/Squares_of_a_Sorted_Array.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    public int[] SortedSquares(int[] A)
    {
        int[] squareA = new int[A.Length];
        for (int i = 0; i < A.Length; ++i)
        {
            squareA[i] = A[i] * A[i];
        }

        Array.Sort(squareA);
        return squareA;
    }

    public int[] str_to_int_array(string s)
    {
        string[] flds = s.Split(',');
        int[] nums = new int[flds.Length];

        if (flds.Length <= 0)
            return nums;

        for (int i = 0; i < nums.Length; ++i)
        {
            nums[i] = int.Parse(flds[i]);
        }

        return nums;
    }

    public string output_int_array(int[] nums)
    {
        if (nums.Length <= 0)
            return "";

        string resultStr = "[" +  nums[0].ToString();
 
        for (int i = 1; i < nums.Length; ++i)
        {
            resultStr += "," + nums[i].ToString();
        }

        return resultStr + "]";
    }
    public void Main(string args)
    {
        string flds = args.Replace("[","").Replace("]","").Trim();
        int[] A = str_to_int_array(flds);

        System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
        sw.Start();

        int[] result = SortedSquares(A);
        Console.WriteLine("result = " + output_int_array(result));

        sw.Stop();
        Console.WriteLine("Execute time ... " + sw.ElapsedMilliseconds.ToString() + "ms\n");
    }
}
